### ISSUE TITLE
Hotspot needs to be turned on twice for it to actually get enabled

### ISSUE THREAD

**Issue by zubozrout** [ubports/ubuntu-touch#1056]
- Device: arale (Meizu MX4)
- Channel: any
- Build: any (at least any Xenial)

### Steps to reproduce
1. Turn on the hotspot (in the indicators area, in the system-settings-app or even using the setup wizard).
2. You may notice it looks as though the hotspot is set-up based on the toggle you just turned on, if not using setup wizard, but it is not. No network was set-up.
3. You can confirm this not being ok by going to the other place - where you have not turned the hotspot on (system-settings-app or indicator area) or leaving the system-settings-app hotspot page and returning to it again. You'll notice the switch is turned off even if you have just tried to set the hotspot up.
4. Toggling the hotspot on again finally turns the hotspot on.

### Expected behavior
Switching the hotspot on only once should be enough.

### Actual behavior
You need to switch the hotspot on twice for it to take effect.

### Additional information
This is a different issue than this one: https://github.com/ubports/ubuntu-touch/issues/281 (this is still the case if I turn the hotspot on twice in the system-setting-app)

I am also experiencing this one: https://github.com/ubports/ubuntu-touch/issues/526 but am not seeing the hotspot being on after using the setup wizard. This does happen all the time - even with the wizard, which indicates a lower-level issue.

This issue has been there for a long time, though I am not sure if it also affected Canonical images or if it is perhaps only related to Xenial.

**Comment by UniversalSuperBox** [comment]
I can confirm this on the Nexus 5. Might be a cross-device issue?

**Comment by zubozrout** [comment]
@UniversalSuperBox Could be though it seems odd this has not been reported yet then. Or maybe it is and I just haven't found it :).

**Comment by Arc676** [comment]
Can also confirm that this happened a few times on Nexus 5 on older builds (don't have a SIM in the device right now, can't check on newest build).

Didn't think to open a new issue since this kind of stuff is basically standard troubleshooting anyway and you almost come to expect requiring a "turn it off and on again" problem on occasion.

**Comment by james-lyon** [comment]
This works as expected on my Aquaris E4.5 (stable) - no need to switch the hotspot on twice.

**Comment by TartanSpartan** [comment]
> Can also confirm that this happened a few times on Nexus 5 on older builds (don't have a SIM in the device right now, can't check on newest build).
> 
> Didn't think to open a new issue since this kind of stuff is basically standard troubleshooting anyway and you almost come to expect requiring a "turn it off and on again" problem on occasion.

In fairness you don't _need_ a SIM to activate the hotspot and test. You can still activate it, it just won't be able to connect to your mobile data connection.

**Comment by Arc676** [comment]
Fair point. Anyway I can confirm that on `2019-06-10` (`devel`) this is still occasionally an issue.

**Comment by TartanSpartan** [comment]
https://pastebin.com/tewNJGwh
https://pastebin.com/LeM6CSTG
https://pastebin.com/YghNHvZ8

This is my /var/log/syslog after a reboot, truncated to that event up to the point where I need to turn hotspot on and off. I split it into three files to meet the Pastebin size restrictions; they are in the correct order. Because of this issue https://github.com/ubports/ubuntu-touch/issues/1160 , I fully activate the hotspot by: turning on hotspot from the indicator menu, going to hotspot settings, hit Change Password/Setup, hit "Require a password" (thereby setting it to green/on), hit Change, and again Change Password/Setup, hit "Require a password" (thereby setting it to grayed out/on), and finally hit Change. When I turn on a WiFi-enabled device, I can now find the Pro 5 hotspot listed among the available connections. Note that before (as I describe in more detail on Telegram for the UBports Supergroup), I had to do this multiple times before it would randomly work, but after deleting my previous networks, it's down to just one like the other users experiencing this bug.

**Comment by libermax** [comment]
With my N5, I can confirm hotspot is touchy :  need to be turn on several times, in general three times. Sometimes but rarely, it works at the first attempt.
I suspected that it might be a race condition between hotspot and wifi activations but never find a way to get it work consistently at the fisrt time, whatever the wait between the activation of wifi and the activation of  hotspot.

I've been using hotspot every day for over two years on my N5. I don't know why, I didn't create a bug report. Perhaps because he has become a companion in everyday life :-) .

**Comment by peterpm** [comment]
With my OPO running OTA-11 I see the following reproducable incorrect behaviour when Wifi is off (I never use Wifi with my phone): 
1. Turn hotspot on: System Settings -> Hotspot: set ON
=> cannot connect with BQ M10 FHD tablet with OTA-11, hotspot name not in list
2. After some 2-3 minutes I check Wifi: System Settings -> Wifi: is ON 
3. Then I go to hotspot again: System Settings -> Hotspot: is OFF
This is wrong, it should be on!
4. Turn hotspot on again while there: Hotspot: set ON
Now the tablet can connect to the phone and the hotspot works without problem.

**Comment by UniversalSuperBox** [comment]
I believe this report is the same as https://github.com/ubports/ubuntu-touch/issues/281 -- please continue any further discussion there.

If I've misread this report or that one, please let me know and I'll reopen this one.

**Comment by zubozrout** [comment]
Well, that's interesting and I never thought of that. I've also already forgot about the older report I've made. You are right both these reports seem closely related but not sure if they are saying the same thing.

I guess the reason #281 is not saying the same thing as #1056 is that I either didn't realize hotspot needs turning on twice to actually turn on or perhaps (and more probably) it turned on right away back on Vivid and only the indicators were broken - while now the indicators are possibly working fine but hotspot is not turned on right away.

And now I see that the description in this ticket also says:
"This is a different issue than this one: #281 (this is still the case if I turn the hotspot on twice in the system-setting-app)"

Anyway, I would probably suggest closing #281 rather than this report as I am no longer able to reproduce #281 (the bit noted in the previous paragraph).

FYI @UniversalSuperBox

**Comment by zubozrout** [comment]
And to add some update this ticket too I no longer seem to be able to enable hotspot solely within the indicator area (no matter how desparately I am trying to toggle that indicator on/off the hotspot icon never appears among the network icons, while System Settings app still requires 2 attempts (or one in the indicator area and then one in the System Settings app) after which hotspot starts working.

**Comment by james-lyon** [comment]
On my Aquaris E4.5 (OTA-15), on the first attempt I can successfully enable the hotspot with 1 attempt. However, if I disable it and enable it again several times, after 2-3 cycles the whole "Network" tab locks up for about 20 seconds. It then becomes responsive again but now I no longer have a wifi connection and it is impossible to reconnect. The wifi slider is green but no networks are found. 

At this point toggling the Hotspot slider causes the Network tab to become unresponsive again.
